src/main/java/org/kohsuke/github/GHEventPayload.java

```diff
@@ -1326,4 +1327,86 @@ void wrapUp(GitHub root) {
             }
         }
     }
+
+    /**
+     * Occurs when someone triggered a workflow run or sends a POST request to the "Create a workflow dispatch event"
+     * endpoint.
+     *
+     * @see <a href=
+     *      "https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#workflow_dispatch">
+     *      workflow dispatch event</a>
+     * @see <a href=
+     *      "https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch">Events that
+     *      trigger workflows</a>
+     */
+    public static class WorkflowDispatch extends GHEventPayload {
+        private Map<String, Object> inputs;
+        private String ref;
+        private String workflow;
+
+        /**
+         * Gets the map of input parameters passed to the workflow.
+         *
+         * @return the map of input parameters
+         */
+        public Map<String, Object> getInputs() {
+            return inputs;
+        }
+
+        /**
+         * Gets the ref of the branch (e.g. refs/heads/main)
+         *
+         * @return the ref of the branch
+         */
+        public String getRef() {
+            return ref;
+        }
+
+        /**
+         * Gets the path of the workflow file (e.g. .github/workflows/hello-world-workflow.yml).
+         *
+         * @return the path of the workflow file
+         */
+        public String getWorkflow() {
+            return workflow;
+        }
+    }
+
+    /**
+     * A workflow run was requested or completed.
+     *
+     * @see <a href=
+     *      "https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#workflow_run">
+     *      workflow run event</a>
+     * @see <a href="https://docs.github.com/en/rest/reference/actions#workflow-runs">Actions Workflow Runs</a>
+     */
+    public static class WorkflowRun extends GHEventPayload {
+        private GHWorkflowRun workflowRun;
+        private GHWorkflow workflow;
+
+        public GHWorkflowRun getWorkflowRun() {
+            return workflowRun;
+        }
+
+        public GHWorkflow getWorkflow() {
+            return workflow;
+        }
```

Comment:
There is also (seems undocumented) a head_sha field that is part of the payload. See https://github.com/redhat-developer/intellij-openshift-connector/blob/master/.github/workflows/publish_reports.yml

This is required for my first Quarkus based GitHub app 🥰

Reply:
It's already there, just nested in the `GHWorkflowRun` object: https://github.com/hub4j/github-api/pull/1064/files#diff-87252b4de52f4e0e1317a91eee8a0de1cc885f7b84f872989c8e52ec09d2509dR41 :)

@jeffmaury did you use https://github.com/quarkiverse/quarkus-github-app ?

Reply:
That's the plan

Reply:
Cool, let me know how it goes :). (and make sure to use the included Replay UI)